### DISCUSSION TITLE
REST GET with body/payload

### DISCUSSION THREAD

**Discussion by agreedSkiing** [quarkusio/quarkus#44616] [Q&A] [answered]
Am i doing something wrong with my test application?

The second endpoint under `/hello/String` should take a body/payload and return it with a modified string, but the swagger-ui tells me this `TypeError: Failed to execute 'fetch' on 'Window': Request with GET/HEAD method cannot have body.` and my request in bash failes silently

```bash
curl -X 'GET' \
  'http://localhost:8080/hello/String' \
  -H 'accept: text/plain' \
  -H 'Content-Type: application/json' \
  -d '{
  "majs": "string",
  "mango": "string",
  "kalle": true,
  "räknare": 0
}'
```
Should it be this way?

[code-with-quarkus.tar.gz](https://github.com/user-attachments/files/17847008/code-with-quarkus.tar.gz)

**Comment by gsmet** [comment]
Why exactly are you trying to post a payload with GET? Because it's not really what you should do. You can pass parameters in the URI but if you want to post an actual payload, you should use POST or PUT.

**Comment by agreedSkiing** [comment]
Mostly because of an internal discussion regarding allowing use of post methods to get data and including concerns regarding maximum length of query parameters when used from a browser, curl, axios or JavaScript fetch.

**Reply by gsmet** [reply, marked as answer]
Well, GET methods are definitely not designed to post a body. I dunno if we support it with Quarkus - it might work - but as you have experienced, a lot of tooling will fail.
I would recommend that you avoid that altogether.

**Comment by agreedSkiing** [comment]
Also since the RFC 9110 states this
 A client SHOULD NOT generate content in a GET request unless it is made directly to an origin server that has previously indicated, in or out of band, that such a request has a purpose and will be adequately supported. An origin server SHOULD NOT rely on private agreements to receive content, since participants in HTTP communication are often unaware of intermediaries along the request chain.

https://www.rfc-editor.org/rfc/rfc9110#section-9.3.1-6

**Comment by dario-sestini** [comment]
Hi, i'm writing a REST client that commuincates with a service that expose a GET endpoint and expects to receive a body. I can't make any change to the remote service as it is not under my control. How can i achieve this? Any suggestion would be appreciated, thanks in advance.

**Reply by gsmet** [reply]
Hey!

If you have a problem and can create a small reproducer allowing us to reproduce the issue, just create an issue in our tracker with the reproducer.

**Reply by dario-sestini** [reply]
Made it, added a code snippet since i think is quite common code and doesn't need a real reproducer. The error come, if i understood it,  from a default behaviour that adheres to the RFC you mentioned above. 
Please refer to this [question](https://stackoverflow.com/questions/79440018/is-there-a-way-to-make-get-request-with-body-using-quarkus-registerrestclient).

**Reply by gsmet** [reply]
What's nice with a reproducer is that we can play with it :).

Maybe try `quarkus-rest-client` instead of `quarkus-resteasy-client` and see how it behaves. But it might be mandated by the Jakarta REST spec and in this case, you won't have any luck.
